src/main/java/ch/njol/skript/lang/function/FunctionReference.java

```diff
@@ -229,17 +231,19 @@ public boolean validateFunction(boolean first) {
 
 		// Check parameter types
 		for (int i = 0; i < parameters.length; i++) {
-			Parameter<?> p = sign.parameters[singleListParam ? 0 : i];
+			Parameter<?> parameter = sign.parameters().get(singleListParam ? 0 : i);
 			RetainingLogHandler log = SkriptLogger.startRetainingLog();
 			try {
+				Class<?> target = Utils.getComponentType(parameter.type());
+
 				//noinspection unchecked
-				Expression<?> e = parameters[i].getConvertedExpression(p.type());
-				if (e == null) {
+				Expression<?> expr = parameters[i].getConvertedExpression(target);
+				if (expr == null) {
 					if (first) {
 						if (LiteralUtils.hasUnparsedLiteral(parameters[i])) {
 							Skript.error("Can't understand this expression: " + parameters[i].toString());
 						} else {
-							String type = Classes.toString(getClassInfo(p.type()));
+							String type = Classes.toString(Classes.getSuperClassInfo(Utils.getComponentType(target)));
```

Comment:
```suggestion
							String type = Classes.toString(Classes.getSuperClassInfo(target));
```
Target already reflects the component type, right?